packages/ember-runtime/lib/system/core_object.js

```diff
@@ -418,7 +435,21 @@ CoreObject.PrototypeMixin = Mixin.create({
     @default false
     @public
   */
-  isDestroying: false,
+  isDestroying: descriptor({
+    get() {
+      let m = meta(this);
+      return (m._flags & SOURCE_DESTROYING) !== 0;
```

Comment:
can we hide the bitmasks behind functions?`m.isSourceDestroying();` I believe it should come at little or no overhead, but be a-bit more maintainable. 

Petka of bluebird fame took this approach, because those methods tend to inline quite nicely. Nice related blog post -> https://reaktor.com/blog/javascript-performance-fundamentals-make-bluebird-fast/